### ISSUE TITLE
Undesired behavior on zoom/pan caused by recent patch

### ISSUE THREAD

**Issue by pearcetm** [openseadragon/openseadragon#2240]
PR https://github.com/openseadragon/openseadragon/pull/2160 has changed the behavior of the viewer to enforce the entire image being visible when the image is zoomed out enough to make this possible. This can cause undesirable panning when zooming out (i.e. if the image is zoomed in, it can be panned off the screen to the extent defined by `visibilityRatio`, but on zooming out, as soon as the image will fit into (a dimension) of the viewer, it will pan to make this happen.

It would be nice if this could be disabled at the very least. It's also not obvious to me why this should be the default behavior - it seems to (conditionally) not respect the `visibilityRatio` parameter. Since respecting `visibilityRatio` when zoomed out was the motivation for the PR that caused this, I'm wondering if this is just a bug and not intended...? I'm not yet certain how to go about changing this back to how it previously worked for the case where I want the zoomed-out image to not be constrained to be within the viewer, while still preserving the improvements for the other use-cases that motivated the PR in the first place.

**Comment by pearcetm** [comment]
Ok, I think I found the problem and a solution. In `Viewport._applyBoundaryConstraints`, the logic introduced by https://github.com/openseadragon/openseadragon/pull/2160 causes inconsistency in behavior based on whether the image is larger or smaller than the viewer window. I think a simpler way to address the original concern from that PR is to use the same logic, but use the smaller of the `(viewer, image)` dimensions in combination with `visibilityRatio` to determine the constraint bounds: 

```
if (newBounds.width > this._contentBoundsNoRotate.width) { 
   horizontalThreshold = this.visibilityRatio * this._contentBoundsNoRotate.width;
} else {
    horizontalThreshold = this.visibilityRatio * newBounds.width;
}
```

Using this approach, for the example of `visibilityRatio = 0.5`, when the image is larger than the viewer, half of the viewer must stay filled; when the image is smaller than the viewer, half of the image must stay in view. At the transition when the image and viewer are the same size, the behavior is the same, which makes zooming behavior in the presence of constraints feel more predictable (in my opinion).

**Comment by iangilman** [comment]
Fixed by #2246.
